keyboards/mechlovin/olly/jf/jf.c

```diff
@@ -0,0 +1,66 @@
+/* Copyright 2021 mechlovin
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "jf.h"
+
+
+void led_init_ports(void) {
+  setPinOutput(C0);
+  setPinOutput(D0);
+  setPinOutput(D1);
+  setPinOutput(C1);
+  setPinOutput(C6);
+  setPinOutput(B0);
+  setPinOutput(B1);
+  setPinOutput(B2);
+}
+
+uint32_t layer_state_set_user(uint32_t state)
+{
+  // if on layer 1, turn on D1 LED, otherwise off.
+    if (biton32(state) == 1) {
+        writePinHigh(D1);
+    } else {
+        writePinLow(D1);
+    }
+  // if on layer 2, turn on D0 LED, otherwise off.
+    if (biton32(state) == 2) {
+        writePinHigh(D0);
+    } else {
+        writePinLow(D0);
+    }
+
+  // if on layer 3, turn on C1 LED, otherwise off.
+    if (biton32(state) == 3) {
+        writePinHigh(C1);
+    } else {
+        writePinLow(C1);
+    }
+  // if on layer 4, turn on C0 LED, otherwise off.
+    if (biton32(state) == 4) {
+        writePinHigh(C0);
+    } else {
+        writePinLow(C0);
+    }
+  // if on layer 5, turn on C6 LED, otherwise off.
+    if (biton32(state) == 5) {
+        writePinHigh(C6);
+    } else {
+        writePinLow(C6);
+    }
+
+    return state;
+}
```

Comment:
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
    writePin(D1, layer_state_cmp(state, 1));
    writePin(D0, layer_state_cmp(state, 2));
    writePin(C1, layer_state_cmp(state, 3));
    writePin(C0, layer_state_cmp(state, 4));
    writePin(C6, layer_state_cmp(state, 5));

    return state;
}
```

Alternatively, if you only want the highest layer.
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
    writePin(D1, get_highest_layer(state) == 1);
    writePin(D0, get_highest_layer(state) == 2);
    writePin(C1, get_highest_layer(state) == 3);
    writePin(C0, get_highest_layer(state) == 4);
    writePin(C6, get_highest_layer(state) == 5);

    return state;
}
```